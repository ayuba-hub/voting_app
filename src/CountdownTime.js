import React, { useEffect, useState } from "react";

const CountdownTime = () => {
  const calculateTimeLeft = () => {
    let year = new Date().getFullYear();
    const difference = +new Date(`${year}-1-22`) - +new Date();
    let timeLeft = {};

    if (difference > 0) {
      timeLeft = {
        days: Math.floor(difference / (1000 * 60 * 60 * 24)),
        hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
        minutes: Math.floor((difference / 1000 / 60) % 60),
        seconds: Math.floor((difference / 1000) % 60),
      };
    }

    return timeLeft;
  };

  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft());
  const [year] = useState(new Date().getFullYear());

  useEffect(() => {
    setTimeout(() => {
      setTimeLeft(calculateTimeLeft());
    }, 1000);
  });

  const timerComponents = [];

  Object.keys(timeLeft).forEach((interval) => {
    if (!timeLeft[interval]) {
      return;
    }

    timerComponents.push(
      <span>
        {timeLeft[interval]} {interval}{" "}
      </span>
    );
  });
  return (
    <div>
      <h1>
        {timerComponents.length ? (
          <span>Voting Ends in: {timerComponents}</span>
        ) : (
          <span>Voting Process has ended!</span>
        )}
      </h1>
    </div>
  );
};

export default CountdownTime;
