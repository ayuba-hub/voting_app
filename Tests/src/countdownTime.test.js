import {shallow} from 'enzyme';
import React from 'react';
import CountdownTime from './CountdownTime';

describe("Test On CountdownTime.js", ()=> {
 
    const container = shallow(<CountdownTime />);

    it('should match the snapshot', () => {
      expect(container.html()).toMatchSnapshot();
    });

    it('expects to render countdown component', () => {
        expect(shallow(<CountdownTime/>).length).toEqual(1)
       
    })
    
    
    it('expects to render countdown component', () => {
        expect(shallow(<CountdownTime/>)).toMatchSnapshot();
    })
    
})
