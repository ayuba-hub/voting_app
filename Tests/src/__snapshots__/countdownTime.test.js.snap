// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test On CountdownTime.js expects to render countdown component 1`] = `ShallowWrapper {}`;

exports[`Test On CountdownTime.js should match the snapshot 1`] = `"<div><h1><span>Voting Process has ended!</span></h1></div>"`;

exports[`Test On Home.js expects to render Home component 1`] = `ShallowWrapper {}`;

exports[`Test On Home.js should match the snapshot 1`] = `"<div><h1><span>Voting Process has ended!</span></h1></div>"`;
